modules/ray_tracing/test/src/userobjects/LotsOfRaysRayStudy.C

```diff
@@ -0,0 +1,282 @@
+      "the normal direction. Must be odd.");
+  params.addRangeCheckedParam<unsigned int>(
+      "azimuthal_quad_order",
+      2,
+      "azimuthal_quad_order > 0",
+      "Order of the azimuthal quadrature per quadrant for angular quadrature generation. The "
+      "azimuthal angle is measured in a plane perpendicular to the normal. Not needed in 2D.");
+
+  params.addParam<bool>(
+      "compute_expected_distance",
+      false,
+      "Whether or not to compute the expected distance for all of the Rays generated");
+
+  // When we define rays, they are already localized on the processors that start them
+  // and the starting elem/incoming side is set
+  params.set<bool>("_claim_after_define_rays") = false;
+  // Can't have replicated Rays if the Rays are on their starting processors only
+  params.set<bool>("_define_rays_replicated") = false;
+  // Don't need to use registration here. For the purposes of testing and this study,
+  // we'll really only ever access the Rays by ID (if at all)
+  params.set<bool>("_use_ray_registration") = false;
+
+  return params;
+}
+
+LotsOfRaysRayStudy::LotsOfRaysRayStudy(const InputParameters & parameters)
+  : RepeatableRayStudyBase(parameters),
+    _vertex_to_vertex(getParam<bool>("vertex_to_vertex")),
+    _centroid_to_vertex(getParam<bool>("centroid_to_vertex")),
+    _centroid_to_centroid(getParam<bool>("centroid_to_centroid")),
+    _edge_to_edge(getParam<bool>("edge_to_edge")),
+    _side_aq(getParam<bool>("side_aq")),
+    _centroid_aq(getParam<bool>("centroid_aq")),
+    _compute_expected_distance(getParam<bool>("compute_expected_distance")),
+    _polar_quad_order(getParam<unsigned int>("polar_quad_order")),
+    _azimuthal_quad_order(getParam<unsigned int>("azimuthal_quad_order"))
+{
+  // Computing the expected distance only works with rectanglar domains because we have to compute
+  // the intersection with the Ray and the domain bounding box (which only represents the domain if
+  // it is rectangular)
+  if (_compute_expected_distance && !isRectangularDomain())
+    paramError("compute_expected_distance", "Not supported when the domain is non-rectangular.");
+
+  const bool polar_set = parameters.isParamSetByUser("polar_quad_order");
+  const bool azimuthal_set = parameters.isParamSetByUser("azimuthal_quad_order");
+  if (_mesh.dimension() == 1)
+  {
+    if (polar_set)
+      paramError("polar_quad_order", "Not needed for 1D");
+    if (azimuthal_set)
+      paramError("azimuthal_quad_order", "Not needed for 1D");
+  }
+  if (!_side_aq && !_centroid_aq)
+  {
+    if (polar_set)
+      paramError("polar_quad_order", "Not needed without side_aq or centroid_aq enabled");
+    if (azimuthal_set)
+      paramError("azimuthal_quad_order", "Not needed without side_aq or centroid_aq enabled");
+  }
+  if (azimuthal_set && _mesh.dimension() == 2)
+    paramError("azimuthal_quad_order", "Not needed for 2D");
```

Comment:
why ? I'd get rid of the polar angle in 2D. Or are you not using the reactor physics convention